internal/controller/management_controller.go

```diff
@@ -759,7 +766,30 @@ func applyKCMDefaults(config *apiextensionsv1.JSON) (*apiextensionsv1.JSON, erro
 	return &apiextensionsv1.JSON{Raw: raw}, nil
 }
 
-func getWrappedComponents(mgmt *kcm.Management, release *kcm.Release) ([]component, error) {
+func applyGlobalRegistry(config *apiextensionsv1.JSON, registry string) (*apiextensionsv1.JSON, error) {
+	values := chartutil.Values{}
+	if config != nil && config.Raw != nil {
+		err := json.Unmarshal(config.Raw, &values)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	globalValues := map[string]any{
+		"global": map[string]any{
+			"registry": registry,
+		},
+	}
+
+	raw, err := json.Marshal(chartutil.CoalesceTables(values, globalValues))
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiextensionsv1.JSON{Raw: raw}, nil
```

Comment:
We already have similar logic in the `enableAdditionalComponents` function. I believe this code block can be moved to this function to avoid extra JSON marshalling and unmarshalling (but the function should probably be renamed).

Reply:
This is optional. I know you wanted to make minimal changes. This can be refactored later.

Reply:
I would suggest adding `TODO:` and `WARN:` or even `kludge` to highlight that it is to be refactored

Reply:
Yes functions `applySveltosDefaults`, `applyKCMDefaults` and `applyGlobalRegistry` could be generalized an put in one place (like in `enableAdditionalComponents`). 
Currently I did it as it was done before in a similar situation. I will put visible comments to all these parts of code and will create a new task for backlog, so it will have a chance to not become permanent.

Reply:
Created #1431